api/authenticator.py

```diff
@@ -2775,6 +2789,10 @@ def get_or_create_token(self, _db, patron):
     def basic_auth_temp_token(self, params, _db):
         """Generate and return a temporary token from HTTP Basic Auth credentials.
         """
+        short_name = self.authenticator.current_library_short_name
+        providers = self.authenticator.library_authenticators[short_name].providers
+        is_new = list(providers)[0].patron_is_new
```

Comment:
Great point, better to err on the side of caution. I'll think of a better approach for this.

Reply:
it's possible that there isn't a super easy way to do it. If there isn't it might be best to just handle the potential `IndexError` and set `is_new` to...false? We'd have to ask OEB what the desired fallback behavior would be

Reply:
I think I've found a better way by filtering for `BasicAuthenticationProvider` then checking that for any patron_is_new